### PR TITLE
[5.0] Cast the $failed variable to object since it can be received as an ar…

### PR DESCRIPTION
**Pull request by fureszpeter**
As soon as we use our custom implementations of a FailedJobProvider, we get an Exception

```
[ErrorException]
Trying to get property of non-object
```

``` php
    /**
     * Get a single failed job.
     *
     * @param  mixed  $id
     * @return array
     */
    public function find($id);
```

We need to cast the array to object and it's working or Interface annotation need to be updated.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please fix your cs.

**Comment by taylorotwell** [comment]
Are you using FETCH_ARRAY or something in your database config?

File: src/Illuminate/Queue/Console/RetryCommand.php [hunk 1]
```diff
@@ -30,6 +30,8 @@ public function fire()
 
 		if ( ! is_null($failed))
 		{
+			$failed = (object)$failed;
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Console/RetryCommand.php]: CS still wrong. `$failed = (object) $failed;`

Comment (GrahamCampbell) [inline src/Illuminate/Queue/Console/RetryCommand.php]: Please squash to one commit after.

**Comment by taylorotwell** [comment]
We will do this on 5.1.

**Comment by GrahamCampbell** [comment]
Thank you @fureszpeter. :)

**Comment by fureszpeter** [comment]
Sorry for the late response...

@taylorotwell 
No, I am not using FETCH_ARRAY. We use custom implementation of **FailedJobProvider**.

ServiceProvider looks like this:
```php
class MongoQueueServiceProvider extends QueueServiceProvider
{
    protected function registerFailedJobServices()
    {
        $this->app->singleton('queue.failer', function ($app) {
            return new MongoDatabaseFailedJobProvider(
                new MongoFailedJobRepository(
                    $this->app->make(MongoDB::class),
                    $this->app->make(FailedJobFactory::class),
                    config('queue.failed')['table']
                ),
                $this->app->make(FailedJobFactory::class),
                $this->app->make(EntityIdFactory::class)
            );
        });
    }
}
```
 **MongoDatabaseFailedJobProvider** implements **FailedJobProviderInterface**

```php
    /**
     * @inheritdoc
     */
    public function find($id)
    {
        $failedJob = $this->repository->getOneById($this->entityIdFactory->make($id));

        if ($failedJob) {
            return (object) $failedJob->toArray();
        } else {
            return null;
        }
    }
```
and **FailedJobProviderInterface**

```php
        /**
         * Get a single failed job.
         *
         * @param  mixed  $id
         * @return array
         */
        public function find($id);
```
So if I implement the interface the code fail, because expecting an object, not an array, this is why I 
suggested ```We need to cast the array to object and it's working or Interface annotation need to be updated.```

**Pull request closed without merging**